### DISCUSSION TITLE
Cannot open xaml file anymore, "Catastrophic failure", VS Avalonia previewer crash?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#13195] [Q&A]
Hi,
Since today, I cannot open XAML files anymore from Visual Studio, I got a Catastrophic failure 0x8000ffff error. Even with new project.

Anyone know how to make it work again?
If I uninstall Avalonia extension, I can open the xaml, but with no previewer, so I think something get wrong with the previewer.

Is there cache files to clean somewhere? or log files?
Thank you.

**Comment by dbriard** [comment]
Right click on a XAML file opened a popup saying than avalonia package do not load correctly with more detail in ActivityLog.xml.

SetSite failed for package [AvaloniaPackage]Source: &apos;AvaloniaVS.VS2022&apos; Description: Could not load file or assembly &apos;Microsoft.VisualStudio.Threading, Version=17.7.0.0

I'll try to upgrate VS to the latest version

**Reply by Takoooooo** [reply]
May you please try and install this extension? Does it work correctly for you? Also test the event completions if possible https://github.com/AvaloniaUI/AvaloniaVS/pull/388
[AvaloniaVS.VS2022 (2).zip](https://github.com/AvaloniaUI/Avalonia/files/12859004/AvaloniaVS.VS2022.2.zip)

**Comment by dbriard** [comment]
Thanks @Takoooooo for your help.
Finally updating Visual Studio fixed the issue :) but I am not really sure what happened.

**Reply by timunie** [reply]
the "tool" to build the extension was updated, and the update required a newer VS version. A fix for older VS will be there soon.

**Reply by Noisyfox** [reply]
> the "tool" to build the extension was updated, and the update required a newer VS version. A fix for older VS will be there soon.

Still get the same issue on old VS with the latest extension.
Seems that I don't have `Microsoft.VisualStudio.Threading, Version=17.4.0.0` on my VS 2022 17.0.32803.143 installation.
